### PR TITLE
feat(compiler.umd): allow root path declaration for Component styleUrls

### PR DESCRIPTION
**Pull request by jonnyasmar**
**Please check if the PR fulfills these requirements**
- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
Need to add a note to https://angular.io/docs/ts/latest/guide/component-styles.html in the note that begins with "The URL is relative to the application root". The new note should read:

> The URL is relative to the application root which is usually the location of the index.html web page that hosts the application. The style file URL is not relative to the component file. That's why the example URL begins app/. See Appendix 2 to specify a URL relative to the component file. Prefix the URL with a `~` to make it relative to the web root (i.e. `~/path/to/styles.css`)

**What kind of change does this PR introduce?** (check one with "x")
```
[X] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)
Resolves this issue: https://github.com/angular/angular/issues/4974
Also corrects a bug with cjs-jasmine that causes a duplicate var declaration in some environments.

**What is the new behavior?**
Component styleUrls can now be prefixed with a `~` to make the URL relative to the web root.

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[X] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
   },
   "dependencies": {
     "core-js": "^2.4.1",
+    "intl": "^1.2.5",
```
Comment (pkozlowski-opensource) [inline package.json]: I don't think this change should be part of this PR....

**Comment by jonnyasmar** [comment]
@pkozlowski-openaource

This feature will not pass the browser test in Travis without updating the version of Intl (recent fix was put in place to address a regex parens matching issue). Sorry, I'm new to github contributing, what would you recommend I do to handle that issue?

**Comment by alfaproject** [comment]
@jonnyasmar that kind of change is better in a separate PR. If you tests fail because of something not related this PR, then it's not an issue.

**Comment by jonnyasmar** [comment]
@pkozlowski-opensource @alfaproject Sorry for the delay!

I've pulled out the intl change and it seems to be passing tests now anyways.

**Comment by jonnyasmar** [comment]
@pkozlowski-opensource @alfaproject bump?

**Comment by DzmitryShylovich** [comment]
@jonnyasmar https://github.com/angular/angular/issues/4974 was fixed so this pr can be closed. thx

**Comment by tbosch** [comment]
Thanks for the PR, however, in 4 and 5 we are not taking new features for JIT mode that will not work in AOT mode.

**Pull request closed without merging**